src/FinFieldsTypes.jl

```diff
@@ -0,0 +1,59 @@
+################################################################################
+#
+#  FinFieldsTypes.jl : some types needed to play with embeddings
+#
+################################################################################
+
+################################################################################
+#
+#  FinFieldsMorphism : embeddings
+#
+################################################################################
+
+struct FinFieldMorphism{T}
```

Comment:
Should belong to Map{D, C, FunctionalMap, FinFieldMorphism}

Reply:
I was wondering if I could use https://github.com/wbhart/AbstractAlgebra.jl/blob/master/src/generic/MapWithInverse.jl#L31-L33 instead of creating a new type. I think this is what I actually need. Would it be ok?

Reply:
That's only for types that have a genuine two sided inverse I think. I suspect you are better off creating a preimage function just like I did in the modules example.

We have to be careful about this because when we come to do category theory, we don't want all the concepts mixed up.